core/unit_test/TestViewCtorProp.hpp

```diff
@@ -0,0 +1,136 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+
+namespace Test {
+
+// Check Kokkos::Impl::is_view_label.
+TEST(TEST_CATEGORY, is_view_label) {
+  static_assert(Kokkos::Impl::is_view_label<std::string>::value);
+
+  constexpr unsigned N = 3;
+  static_assert(Kokkos::Impl::is_view_label<const char[N]>::value);
+  static_assert(Kokkos::Impl::is_view_label<char[N]>::value);
+
+  static_assert(!Kokkos::Impl::is_view_label<char*>::value);
+}
+
+// Check traits of Kokkos::Impl::ViewCtorProp<void, std::string>.
+TEST(TEST_CATEGORY, vcp_label_base_traits) {
+  using vcp_label_base_t = Kokkos::Impl::ViewCtorProp<void, std::string>;
+
+  static_assert(std::is_default_constructible_v<vcp_label_base_t>);
+  static_assert(std::is_copy_constructible_v<vcp_label_base_t>);
+  static_assert(std::is_copy_assignable_v<vcp_label_base_t>);
+  static_assert(std::is_move_constructible_v<vcp_label_base_t>);
+  static_assert(std::is_move_assignable_v<vcp_label_base_t>);
```

Comment:
Would it be worth adding a single trait that does all these checks ? (at least for this test)

Reply:
We have shortened the tests. The tests of the traits are now focused on testing the properties of being "default constructible" and "copy constructible" for the specific cases that are needed for existing use cases in Kokkos. We deleted the tests of the other traits.